cpp/include/raft/neighbors/nn_descent_types.hpp

```diff
@@ -100,12 +109,18 @@ struct index : ann::index {
    * @param graph_view raft::host_matrix_view<IdxT, int64_t, raft::row_major> for storing knn-graph
    */
   index(raft::resources const& res,
-        raft::host_matrix_view<IdxT, int64_t, raft::row_major> graph_view)
+        raft::host_matrix_view<IdxT, int64_t, raft::row_major> graph_view,
+        std::optional<raft::host_matrix_view<DistData_t, int64_t, raft::row_major>> distances_view =
```

Comment:
@cjnolet and I discussed this offline, let's keep the distance matrix as `raft::device_matrix_view`